cypress/e2e/catalog/giftCards/exportGiftCards.js

```diff
@@ -0,0 +1,74 @@
+/// <reference types="cypress" />
+/// <reference types="../../../support"/>
+
+import faker from "faker";
+
+import { GIFT_CARD_LIST } from "../../../elements/catalog/giftCard/giftCardList";
+import { GIFT_CARD_SHOWMORE } from "../../../elements/catalog/giftCard/giftCardShowMore";
+import { ASSIGN_ELEMENTS_SELECTORS } from "../../../elements/shared/assign-elements-selectors.js";
+import { BUTTON_SELECTORS } from "../../../elements/shared/button-selectors";
+import { TEST_ADMIN_USER } from "../../../fixtures/users";
+import { createGiftCard } from "../../../support/api/requests/GiftCard";
+import { deleteGiftCardsWithTagStartsWith } from "../../../support/api/utils/catalog/giftCardUtils";
+import { getMailWithGiftCardExport } from "../../../support/api/utils/users";
+import { enterAndSelectGiftCards } from "../../../support/pages/catalog/giftCardPage";
+
+describe("As an admin I want to export gift card", () => {
+  const startsWith = "updateGCard";
+
+  before(() => {
+    cy.clearSessionData().loginUserViaRequest();
+    deleteGiftCardsWithTagStartsWith(startsWith);
+  });
+
+  beforeEach(() => {
+    cy.clearSessionData().loginUserViaRequest();
+  });
+
+  it(
+    "should be able to export several gift cards. TC: SALEOR_1010",
+    { tags: ["@giftCard", "@allEnv", "@stable"] },
+    () => {
+      const giftCard01 = `${startsWith}${faker.datatype.number()}`;
+      const giftCard02 = `${startsWith}${faker.datatype.number()}`;
+      let giftCard01hash;
+      let giftCard02hash;
+
+      createGiftCard({
+        tag: giftCard01,
+        amount: 5,
+        currency: "THB",
+      })
+        .then(hash => {
+          giftCard01hash = hash.id;
+          createGiftCard({
+            tag: giftCard02,
+            amount: 10,
+            currency: "THB",
+          });
+        })
+        .then(hash2 => {
+          giftCard02hash = hash2.id;
+          enterAndSelectGiftCards([giftCard01hash, giftCard02hash]);
+          cy
+            .get(ASSIGN_ELEMENTS_SELECTORS.checkbox)
+            .first()
+            .check()
+            .should("be.checked")
+            .get(GIFT_CARD_LIST.selectedAmount)
+            .contains("Selected 2 items")
+            .should("be.visible")
+            .get(BUTTON_SELECTORS.showMoreButton)
+            .click()
+            .get(GIFT_CARD_SHOWMORE.exportCodesMenu)
+            .click()
+            .get(BUTTON_SELECTORS.submit)
+            .click().confirmationMessageShouldDisappear;
+          getMailWithGiftCardExport(
+            TEST_ADMIN_USER.email,
+            "Your exported gift cards data is ready",
+          );
```

Comment:
This will return all emails with export, for the admin user. If there is an email from yesterday, and today this email won't be sent, this test still will pass. Imagine that we are running these tests for all versions at the same time, and on one of the versions this is broken.

Reply:
I think the best solution will be to change the title of this email to some random value before this test via API (configuration -> plugins -> admin emails)